wallet/src/main/java/io/neow3j/wallet/Wallet.java

```diff
@@ -142,6 +186,11 @@ public boolean addAccount(Account account) {
      * @return true if an account was removed, false if no account with the given address was found.
      */
     public boolean removeAccount(ScriptHash scriptHash) {
+        if (scriptHash == this.getDefaultAccount().getScriptHash()
+                && this.accounts.size() > 1) {
+            ScriptHash newDefaultAccount = this.accounts.entrySet().iterator().next().getKey();
+            this.setDefaultAccount(this.accounts.get(newDefaultAccount).getScriptHash());
+        }
         return accounts.remove(scriptHash) != null;
```

Comment:
- Need to remove the account's reference to the wallet from the account (i.e. set to null).
- Need to clear the wallet's `defaultAccount` variable if the last account is removed (i.e. set to null)
- Extend corresponding unit test to cover this behavior.